sdk/python/kfp/client/client.py

```diff
@@ -1082,7 +1081,7 @@ def create_run_from_pipeline_func(
     def create_run_from_pipeline_package(
         self,
         pipeline_file: str,
-        arguments: Optional[Dict[str, str]] = None,
```

Comment:
Is it possible to submit a run with `arguments={'number': 1}`?

Reply:
I’m able to get the function to execute even when the `arguments` dictionary includes an int. [This](https://www.kubeflow.org/docs/components/pipelines/v2/author-a-pipeline/component-io/#inputs-1) documentation indicates that non-string values are an accepted pattern.

> Valid input parameter annotations include str, int, float, bool, dict, list.

Reply:
In that case, perhaps the original annotation is correct?

Reply:
Also, these two functions just wrap `run_pipeline`, which uses the `Dict[str, any]` syntax in [its type hints](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L736).

`run_pipeline` invokes `_create_job_config`, which also uses [Any](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L736),

Reply:
After a few more hops, it eventually winds up at [this](https://github.com/kubeflow/pipelines/blob/master/backend/api/v1beta1/go_http_client/job_model/pipeline_spec_runtime_config.go#L21) Go struct, which has a value of `interface{}`, i.e. an empty interface, which matches _all_ types.

Reply:
Just to clarify and add some context, [create_run_from_pipeline_func](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L1027) in this _same_ module uses `Dict[str, Any]` as opposed to `Dict[str, str]`. So two functions in the same module that essentially do the same thing have different type hints. If that's not intentional, then they should probably be brought into alignment, which begs the question, which one is right? Since they both wrap around functions (and ultimately types) that use `Dict[str, Any]`, it seems like that's the standard and `create_run_from_pipeline_package` just has a small discrepancy that needs to be brought into alignment. Maybe I'm missing something though. As my wife can attest, I'm wrong all the time.

Reply:
Thanks for explaining. I misunderstood :). This change LGTM!

Reply:
I believe the TODO comment is about type validation, however (i.e., make sure the values submitted are the correct types based on the pipeline signature), not about the type annotation. Can you re-add the comment?

Reply:
Added the comments back! Thanks, @connor-mccarthy!

Reply:
Can you also update the title since the comments aren't removed? Also, since the TODOs are unrelated, this is a fairly small PR. Are there other needed `client.py` type annotation updates that we can bundle in to this PR?

Reply:
Good call. Happy to tackle that.

Reply:
Updated about 20 type hints and modified the commit and PR title. Happy to address any other feedback.